exporter/internal/queue/persistent_queue.go

```diff
@@ -340,45 +341,49 @@ func (pq *persistentQueue[T]) getNextItem(ctx context.Context) (T, func(error),
 			pq.logger.Error("Error deleting item from queue", zap.Error(err))
 		}
 
-		return request, nil, false
+		return index, request, false
```

Comment:
nit: just to stay consistent and return zero values when unsuccessful 
```suggestion
		return 0, request, false
```